securedrop/tests/test_integration.py

```diff
@@ -17,223 +17,236 @@
 os.environ['SECUREDROP_ENV'] = 'test'  # noqa
 import config
 import crypto_util
-from db import db_session, Journalist
-import journalist
-import source
+from db import db
+from models import Journalist
+import journalist_app
+import source_app
 import store
 import utils
 
 
 class TestIntegration(unittest.TestCase):
 
-    def _login_user(self):
-        self.journalist_app.post('/login', data=dict(
-            username=self.user.username,
+    def _login_user(self, app):
+        app.post('/login', data=dict(
+            username=self.username,
             password=self.user_pw,
             token='mocked'),
             follow_redirects=True)
 
     def setUp(self):
-        utils.env.setup()
+        self.source_app = source_app.create_app(config)
+        self.journalist_app = journalist_app.create_app(config)
 
-        self.source_app = source.app.test_client()
-        self.journalist_app = journalist.app.test_client()
+        self.__context = self.journalist_app.app_context()
+        self.__context.push()
+        utils.env.setup()
 
         self.gpg = gnupg.GPG(homedir=config.GPG_KEY_DIR)
 
         # Patch the two-factor verification to avoid intermittent errors
-        patcher = mock.patch('db.Journalist.verify_token')
+        patcher = mock.patch('models.Journalist.verify_token')
         self.addCleanup(patcher.stop)
         self.mock_journalist_verify_token = patcher.start()
         self.mock_journalist_verify_token.return_value = True
 
         # Add a test user to the journalist interface and log them in
         # print Journalist.query.all()
         self.user_pw = "corret horse battery staple haha cultural reference"
-        self.user = Journalist(username="some-username",
-                               password=self.user_pw)
-        db_session.add(self.user)
-        db_session.commit()
-        self._login_user()
+        self.username = crypto_util.genrandomid()
```

Comment:
I'm fine with this, but if we do then we should specify the random seed for deterministic unit tests runs (see [here](https://github.com/freedomofpress/securedrop/blob/develop/securedrop/tests/test_journalist.py#L25))